### ISSUE TITLE
[Refactoring]: Fix the hanging PR test runs

### ISSUE THREAD

**Issue by kwvanderlinde** [RPTools/maptool#3750]
### Describe the problem

For the last several PRs I submitted, I've noticed that the Windows and Mac checks often run stuck and need to be cancelled and rerun. Sometimes this needs to be done multiple times before a successful build completes.

### The improvement you'd like to see

The tests should run reliably so that the automated PR checks can be completed in a reasonable time frame.

### Expected Benefits

We can rely on the automated checks to bring value to the PR process, rather than just holding things up as a formality.

### Additional Context

The checks always seems to hang in gradle's `:test` task.

**Comment by kwvanderlinde** [comment]
I haven't had the chance to directly test on Windows (and I don't have a Mac), so for now I have to settle for testing on Linux. So my findings might not be the same problem as seen in the PRs, but it's stuff that needs fixing anyways.

I've found that `CampaignPropertiesDialogTest` can sometimes deadlock during the dialog construction. Specifically, the main thread hangs deep inside the `pack()` call, waiting on a reader lock that is blocked by the AWT event queue thread's hold on a writer. Meanwhile, the AWT event queue thread is stuck waiting on the global `Component.LOCK`.

I had success changing this test to run on the Swing thread instead of the main test thread (`SwingUtilities.invokeLater()`), which I think is the right way to avoid the deadlock. But I say that as someone who know next to nothing about the details of Swing, so if there are any Swing experts that want to correct me, please do.

There are also a couple other minor things I've found which I'll also include in a PR, but they are much more rare and don't result in deadlocks.

**Comment by Azhrei** [comment]
That’s likely the issue…

Anything related to the GUI or typical event processing should be done on the Swing thread. MT does have some use of other threads for the A* routing, a rerendering throttle (IIRC), and probably a few other things, which will all conspire to trip people up when writing tests.

I’m not familiar with that/those tests, but I have to wonder why the GUI needs to be constructed in the first place. That class should probably delegate to another class/object for implementation details (Strategy pattern, I think?) and that’s the one that should be tested… (Correct me if I’m wrong…!)

**Comment by kwvanderlinde** [comment]
> That class should probably delegate to another class/object for implementation details ...

If only our logic was so well separated :smile:

**Comment by kwvanderlinde** [comment]
Oh, I should mention how I went about diagnosing the issue, because I had no idea this existing until today.

In IntelliJ, I added a JUnit configuration to the Run/Debug Configurations, set it to run all tests under `src/test/java/`, and then set the option to [run repeatedly](https://www.jetbrains.com/help/idea/run-debug-configuration-junit.html#tests) until it encounters a failure. This let me easily do thousands of test runs while looking for and debugging the problem. Without the repetition, it would have been a real hassle considering that the issue was not consistently presenting itself.
